python/modem-firmware-1.3+/command_interface.py

```diff
@@ -166,3 +178,88 @@ def get_mfw_version(self):
+    def write_credential(self, sectag, cred_type, cred_text):
+        # Because the Zephyr shell does not support multi-line commands,
+        # we must base-64 encode our PEM strings and install them as if they were binary.
+        # Yes, this does mean we are base-64 encoding a string which is already mostly base-64.
+        # We could alternatively strip the ===== BEGIN/END XXXX ===== header/footer, and then pass
+        # everything else directly as a binary payload (using BIN mode instead of BINT, since
+        # MBedTLS uses the NULL terminator to determine if the credential is raw DER, or is a
+        # PEM string). But this will fail for multi-CA installs, such as CoAP.
+
+        # text -> bytes -> base64 bytes -> base64 text
+        encoded = base64.b64encode(cred_text.encode()).decode()
+
+        # Clear credential buffer -- If it is already clear, there may not be text feedback
+        self.write_raw("cred buf clear")
+
+        # Write the encoded credential in chunks
+        chunks = math.ceil(len(encoded)/TLS_CRED_CHUNK_SIZE)
+        for c in range(chunks):
+            chunk = encoded[c*TLS_CRED_CHUNK_SIZE:(c+1)*TLS_CRED_CHUNK_SIZE]
+            self.write_raw(f"cred buf {chunk}")
+            self.serial_wait_for_response("Stored")
+
+        # Store the buffered credential
+        self.write_raw(f"cred add {sectag} {TLS_CRED_TYPES[cred_type]} DEFAULT bint")
+        result, output = self.serial_wait_for_response("Added TLS credential")
+        time.sleep(1)
+        return result
+
+    def delete_credential(self, sectag, cred_type):
+        self.write_raw(f'cred del {sectag} {TLS_CRED_TYPES[cred_type]}')
+        result, output = self.serial_wait_for_response("Deleted TLS credential", "There is no TLS credential")
+        time.sleep(2)
+        return result
+
+    def check_credential_exists(self, sectag, cred_type, get_hash=True):
+        self.write_raw(f'cred list {sectag} {TLS_CRED_TYPES[cred_type]}')
+
+        # This will capture the list dump for the credential if it exists.
+        result, output = self.serial_wait_for_response("1 credentials found.",
+                                                       "0 credentials found.",
+                                                       store=
+                                                       f"{sectag},{TLS_CRED_TYPES[cred_type]}")
+        if result and output:
+            if not get_hash:
+                return True, None
+
+            # Output is a comma separated list of positional items
+            data = output.decode().split(",")
+            hash = data[2].strip()
+            status_code = data[3].strip()
+
+            if (status_code != "0"):
+                print(error_style(f"Error retrieving credential hash: {output.decode().strip()}."))
+                print(error_style("Device might not support credential digests."))
+                return True, None
+
+            return True, hash
+
+        return False, None
```

Comment:
On second glance, I think the if-statement depth could be reduced with some inversions